projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_router.py

```diff
@@ -59,6 +64,17 @@ def send_object_for_ingestion(
         target: Optional[str] = None,
         collection_id: Optional[str] = None,
     ):
+        # Initialize the pre- and post- processors.
+        if not self._initialized_pre_processors:
```

Comment:
why initialising processors on each send_*_for_ingestion?
why not when initialising the ingesterbase in `__initialize_ingester()` where they are used?

Reply:
@ivakoleva 
They are not initialized on each send_*_for_ingestion, they are initialized only on the first invocation of a send_*_for_ingestion method, and then reused.
Moving the initialization to where the IngesterBase is initialized does not seem to provide any significant benefit.